src/screencast/xdpw_gbm.c

```diff
@@ -0,0 +1,56 @@
+#include <gbm.h>
+#include <xf86drm.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <string.h>
+
+#include "xdpw.h"
+#include "logger.h"
+
+static char *gbm_find_render_node(size_t maxlen) {
+	drmDevice *devices[64];
+	char *render_node = NULL;
+
+	int n = drmGetDevices2(0, devices, sizeof(devices) / sizeof(devices[0]));
+	for (int i = 0; i < n; ++i) {
+		drmDevice *dev = devices[i];
+		if (!(dev->available_nodes & (1 << DRM_NODE_RENDER)))
+			continue;
+
+		render_node = strndup(dev->nodes[DRM_NODE_RENDER], maxlen);
+		break;
+	}
+
+	drmFreeDevices(devices, n);
+	return render_node;
+}
+
+struct gbm_device *create_gbm_device(){
+	struct gbm_device *gbm;
+	char *render_node = NULL;
+	size_t render_node_size = 256;
+
+	render_node = gbm_find_render_node(render_node_size);
+	if (render_node == NULL) {
+		logprint(ERROR, "xdpw: Could not find render node");
+		return NULL;
+	} else {
+		logprint(INFO, "xdpw: Using render node %s",render_node);
+	}
+
+	int fd = open(render_node, O_RDWR);
```

Comment:
Should use `O_CLOEXEC` too to avoid leaking the FD to child processes.

Reply:
Can we ignore closing the fd on shutdown with this flag?

Reply:
This flag means "close on exec", it means the FD will be automatically closed when performing an exec(2) syscall such as we do when spawning the chooser. The flag doesn't change what happens on shutdown.

It's a good practice to close the FDs as necessary, that way FD leaks are less annoying to track down. Also if we ever support features like config hot-reloading, closing at the correct time avoids leaving many FDs open.

Reply:
Ah read it as "close on exit", which thinking about it doesn't make any sense.